main.go

```diff
@@ -1,9 +1,482 @@
+
+	// builds a slice containing the theoretical distances
+	// inputs : s = slice of the zcmb, n = number of steps for the integral, to be passed to the integral function through mu_th
+
+	diag_ord := make([]float64, 0)
+
+	for i := 0; i < len(s); i++ {
+		diag_ord = append(diag_ord, mu_th(s[i], n))
+	}
+
+	return diag_ord
+}
+
+/*
+######
+Experimental mu
+######
+*/
+
+func mu_exp_slice(m_stell, mb, stretch, colour []float64) []float64 {
+
+	// compute experimental mus and add them in an array
+	// inputs : mB = B band peak magnitude, stretch and colour = SALT2 parameters
+
+	slice := make([]float64, 0)
+	for i := 0; i < len(mb); i++ {
+		if m_stell[i] < 10 {
+			val := mb[i] - Mb + alpha*stretch[i] - beta*colour[i]
+			slice = append(slice, val)
+		} else {
+			val := mb[i] - Mb - delta_M + alpha*stretch[i] - beta*colour[i]
+			slice = append(slice, val)
+		}
+	}
+
+	return slice
+}
+
+func mu_exp_err(dmb, dstretch, dcolour []float64) []float64 {
+
+	// compute errors on experimental mus and add them in an array
+	// inputs : dmb, dstretch, dcolour = errors on mb, stretch and colour from JLA data
+
+	slice := make([]float64, 0)
+	for i := 0; i < len(dmb); i++ {
+		val := math.Sqrt(math.Pow(dmb[i], 2) + math.Pow(alpha*dstretch[i], 2) + math.Pow(beta*dcolour[i], 2))
+		slice = append(slice, val)
+	}
+
+	return slice
+}
+
+
+/*
+######
+Functions for diagrams : plots, fit, etc
+######
+*/
+
+func points(x_values, y_values []float64) plotter.XYs {
+
+	// creates a plotter.XYs (cloud of points) from given 2D coordinates
+	// inputs : x_values, y_values = slices containing the x and y values respectively
+
+	pts := make(plotter.XYs, len(x_values))
+	for i := range pts {
+		pts[i].X = x_values[i]
+		pts[i].Y = y_values[i]
+	}
+
+	return pts
+}
+
+func diag(points plotter.XYs) *plot.Plot {
+
+	// plot the given points
+	// input : points = plotter.XYs created by the points function
+
+	p, err := plot.New()
+	if err != nil {
+		log.Panic(err)
+	}
+
+	p.Title.Text = "Hubble diagram"
+	p.X.Label.Text = "Redshift z"
+	p.Y.Label.Text = "Distance mu"
+	p.X.Scale = plot.LogScale{}
+	p.X.Tick.Marker = plot.LogTicks{}
+	p.Add(plotter.NewGrid())
+
+	s, err := plotter.NewScatter(points)
+	if err != nil {
+		log.Panic(err)
+	}
+
+	s.GlyphStyle.Color = color.RGBA{R: 255, B: 0, A: 255}
+	s.GlyphStyle.Radius = vg.Points(3)
+	s.GlyphStyle.Shape = draw.PlusGlyph{}
+
+	p.Add(s)
+
+	return p
+}
+
+func modified_integral(z, omega float64) float64 {
+
+	// modified integral function where omgM is a variablef that can be optimized
+	// inputs : z = redshift, omega = omgM as a free parameter
+
+	var inte float64
+	abs := make([]float64, 0)
+	ord := make([]float64, 0)
+
+	var i float64
+	for i = 0; i <= 1000; i++ {
+		val := z * i / 1000
+		abs = append(abs, val)
+	}
+
+	for j := 0; j <= 1000; j++ {
+		val := 1 / math.Sqrt((1+abs[j]*omega)*(1+abs[j])*(1+abs[j])-(1-omega)*(2+abs[j])*abs[j])
+		ord = append(ord, val)
+	}
+
+	inte = integrate.Trapezoidal(abs, ord)
```

Comment:
just return it directly without using this temporary variable that you had trouble naming :)

```go
return integrate.Trapezoidal(abs, ord)
```